evalml/tests/pipeline_tests/test_time_series_baseline_pipeline.py

```diff
@@ -73,3 +74,52 @@ def test_time_series_baseline(
     importance = np.array([0] * transformed.shape[1])
     importance[delay_index] = 1
     np.testing.assert_allclose(clf.estimator.feature_importance, importance)
+
+
+@pytest.mark.parametrize("forecast_horizon,gap", [[3, 0], [10, 2], [2, 5]])
+@pytest.mark.parametrize("numeric_idx", [True, False])
+def test_time_series_get_forecast_periods(forecast_horizon, gap, numeric_idx, ts_data):
+    X, _, y = ts_data(problem_type=ProblemTypes.TIME_SERIES_REGRESSION)
+    if numeric_idx:
+        X = X.reset_index(drop=True)
+    clf = make_timeseries_baseline_pipeline(
+        ProblemTypes.TIME_SERIES_REGRESSION,
+        gap,
+        forecast_horizon,
+        time_index="date",
+    )
+
+    with pytest.raises(
+        ValueError,
+        match="Pipeline must be fitted before getting forecast.",
+    ):
+        clf.get_forecast_periods(X)
+
+    clf.fit(X, y)
+    result = clf.get_forecast_periods(X)
+
+    assert result.size == forecast_horizon + gap
+    assert all(result.index == range(len(X), len(X) + forecast_horizon + gap))
+    assert result.iloc[0] == X.iloc[-1]["date"] + np.timedelta64(1, "D")
+    assert np.issubdtype(result.dtype, np.datetime64)
+    assert result.name == "date"
+
+
+@pytest.mark.parametrize("forecast_horizon,gap", [[3, 0], [10, 2], [2, 5]])
+def test_time_series_get_forecast_predictions(forecast_horizon, gap, ts_data):
+    X, _, y = ts_data(problem_type=ProblemTypes.TIME_SERIES_REGRESSION)
+
+    X_train, y_train = X.iloc[:15], y.iloc[:15]
+    X_validation = X.iloc[15 : (15 + gap + forecast_horizon)]
+
+    clf = make_timeseries_baseline_pipeline(
+        ProblemTypes.TIME_SERIES_REGRESSION,
+        gap,
+        forecast_horizon,
+        time_index="date",
+    )
+    clf.fit(X_train, y_train)
+    forecast_preds = clf.get_forecast_predictions(X=X_train, y=y_train)
+    X_val_preds = clf.predict(X_validation, X_train=X_train, y_train=y_train)
+
+    assert_series_equal(forecast_preds, X_val_preds)
```

Comment:
Check my understanding here: the only reason these two predictions are the same is because we're using a baseline pipeline, so the extra feature in X doesn't have an impact. For other models though, this isn't going to be the case.

Can we add info/a warning in the function (and docstring, as well as tests) to make users aware that the `get_forecast_predictions` function only works for univariate problems? 

Is that something we're ok with, since we're forecasting?

Reply:
These forecasting functions are meant to work for both univariate and multivariate problems. Because we have a forecast horizon and a gap with time series problems, we need to pull from past data. In the case of multivariate problems, we are pulling these features from past rows of `X_train` after it is transformed through the entire pipeline. For univariate problems, just the periods generated should be fine. 

I agree that we should probably use a more complex pipeline that utilizes features though. Updated to include this.